public/app/pipeline-orchestrator.js

```
@@ -414,20 +350,87 @@ class PipelineOrchestrator {
         resolve(); // non-fatal — DSP passthrough still runs
       };
 
+      // ── Apply graceful-degradation patch (ml-worker-models-patch.js) ────
+      // Must come AFTER onmessage is set so the patch wraps the real handler.
+      // Stamps ⚠ DSP badges on pipeline stage UI elements for any missing
+      // .onnx files. Non-destructive — worker still functions without models.
+      if (typeof window._mlWorkerPatch === 'function') {
```

ESLint

[error] 'window' is not defined.

(no-undef)